### PR TITLE
fixes 2-456: Preserve all data from strategy import

### PR DESCRIPTION
**Pull request by chriswk**
## What
Previously when importing strategies we've used the same data type we've used for creating strategies (the minimal, a name, an optional description, optional parameters and an optional editable column). This meant that exporting strategies and then importing them would reactivate deprecated strategies. This PR changes to allow the import to preserve all the data coming in the export file.

## Tests
Added four new tests, two new unit tests using our fake stores and two new e2e tests. Interestingly the ones in the fake store ran green before this change as well, probably because we just insert the parsed json object in the fake store, whereas the real store actually converts the object from camelCasing to the postgresql snake_casing standard.  

### Mismatch between fake and real stores
This is inevitable since storing things in javascript arrays vs saving in a real database will have some differences, but this again shows the value of our e2e tests.

### Invariants
Should we see if we can add some invariants to our import/export so that we can write some proptests for it? One candidate is commutativity of import/export. On a fresh database importing and then exporting should yield the same file that was imported provided all flags are turned on. Candidate for Q1 improvement of import/export.

### CODE REVIEW

File: src/lib/db/strategy-store.ts [hunk 1]
```diff
@@ -157,8 +158,16 @@ export default class StrategyStore implements IStrategyStore {
     }
 
     // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-    async importStrategy(data): Promise<void> {
-        const rowData = this.eventDataToRow(data);
+    async importStrategy(data: IStrategyImport): Promise<void> {
```
Comment (gastonfournier) [inline src/lib/db/strategy-store.ts:161]: You can now remove the line above:
`// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types`

**Review by gastonfournier — COMMENTED** [review]
Looks good, just 2 small comments

File: src/lib/types/stores/strategy-store.ts [hunk 2]
```diff
@@ -23,6 +23,16 @@ export interface IMinimalStrategy {
     parameters?: any[];
 }
 
+export interface IStrategyImport {
```
[the remark this answers is not in the document]
Comment (chriswk) [reply src/lib/types/stores/strategy-store.ts:26]: Yeah, we call it builtIn in the export as well as built_in in the table. where editable is `built_in === 0`. Unfortunately built_in is still an integer in our table for some reason :)

[the remark this answers is not in the document]
Comment (gastonfournier) [reply src/lib/types/stores/strategy-store.ts:26]: Oh, I just checked the previous code using 
https://github.com/Unleash/unleash/blob/be045dc13a9aefc4bdcf0d87d3707b44dcbfad70/src/lib/db/strategy-store.ts#L127-L133

and I see that `editable` although present in the type was completely ignored :+1:

**Review by gastonfournier — APPROVED** [review]

File: src/lib/types/stores/strategy-store.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (chriswk) [reply src/lib/types/stores/strategy-store.ts:26]: yup :(

**Pull request merged**